File: Example4OC/Example4OC/Utils/BVRegular.m
//
//  BVRegular.m
//  LibosExample
//
//  Created by 黑眸智能 on 2021/11/2.
//

#import "BVRegular.h"
#import <GZIP.h>

@implementation BVRegular

+(NSData *)loadData {
    NSURL *bundle = [[NSBundle mainBundle] URLForResource:@"json" withExtension:@"txt"];
    NSData *data = [NSData dataWithContentsOfURL:bundle];
    NSData *gData = [data gzippedData];
    return gData;
}

+(NSString *)loadJsonString {
    NSURL *bundle = [[NSBundle mainBundle] URLForResource:@"map" withExtension:@"txt"];
    NSString *str = [NSString stringWithContentsOfURL:bundle encoding:NSUTF8StringEncoding error:nil];
    return str;
}

@end

File: Example4Swift/Example4Swift/Utils/BVRegular.swift
//
//  BVRegular.swift
//  Libos
//
//  Created by 黑眸智能 on 2021/10/13.
//

import UIKit
import GZIP

class BVRegular: NSObject {
    class func loadData() -> Data?{
        if let bundle = Bundle.main.url(forResource: "compr_map", withExtension: "txt") {
            do {
                let data = try Data.init(contentsOf: bundle)
                if let gData = (data as NSData).gzipped() {
                    return gData
                }
            } catch let error {
                print("出错:\(error)")
            }
        }
        
        return nil
    }
}
